core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -1828,6 +1814,35 @@ impl<R: WriteableRocks> StateTreeGcStore for StateManagerDatabase<R> {
             .iterate(Direction::Forward)
     }
 
+    fn progress_historical_substate_values_availability(&self, available_from: StateVersion) {
+        let db_context = self.open_rw_context();
+        let extension_data_cf = db_context.cf(ExtensionsDataCf);
+        let current_available_from = extension_data_cf
+            .get(&ExtensionsDataKey::StateTreeAssociatedValuesStatus)
+            .map(|bytes| {
+                scrypto_decode::<VersionedStateTreeAssociatedValuesStatus>(&bytes)
+                    .unwrap()
+                    .into_latest()
+                    .historical_substate_values_available_from
+            });
+        let Some(current_available_from) = current_available_from else {
+            // The state history feature is simply not enabled.
+            return;
+        };
+        if available_from <= current_available_from {
+            // The state history feature was enabled after this state version.
+            return;
+        }
+        let updated_status =
```

Comment:
Minor note - but this code would be slightly safer with respect to possible changes (e.g. new versions of status, or addition of other properties to the status) if we take:
```rust
        let mut status = extension_data_cf
            .get(&ExtensionsDataKey::StateTreeAssociatedValuesStatus)
            .map(|bytes| {
                scrypto_decode::<VersionedStateTreeAssociatedValuesStatus>(&bytes)
                    .unwrap()
                    .into_latest()
            });
// ...
        status.historical_substate_values_available_from = available_from;
        extension_data_cf.put(
            &ExtensionsDataKey::StateTreeAssociatedValuesStatus,
            &scrypto_encode(&VersionedStateTreeAssociatedValuesStatus::from(status)).unwrap(),
        );
    
```

Reply:
good idea, done 👍